lens-cube/src/main/java/org/apache/lens/cube/parse/MaxCoveringFactResolver.java

```diff
@@ -31,193 +30,174 @@
+        if (!partCountsPerPartCol.containsKey(entry.getKey())) {
+          partCountsPerPartCol.put(entry.getKey(), Maps.<Candidate, Long>newHashMap());
+        }
+        partCountsPerPartCol.get(entry.getKey()).put(cand, entry.getValue());
+      }
+    }
+    // for each queried partition, prune fact sets that are covering less range than max
+    for (String partColQueried : cubeql.getPartitionColumnsQueried()) {
+      if (partCountsPerPartCol.get(partColQueried) != null) {
+        long maxTimeCovered = Collections.max(partCountsPerPartCol.get(partColQueried).values());
+        TimeCovered timeCovered = new TimeCovered(maxTimeCovered);
+        Iterator<Candidate> iter = cubeql.getCandidates().iterator();
+        while (iter.hasNext()) {
+          Candidate candidate = iter.next();
+          Long timeCoveredLong = partCountsPerPartCol.get(partColQueried).get(candidate);
+          if (timeCoveredLong == null) {
+            timeCoveredLong = 0L;
+          }
+          if (timeCoveredLong < maxTimeCovered) {
+            log.info("Not considering Candidate:{} from Candidate set as it covers less time than the max"
+              + " for partition column: {} which is: {}", candidate, partColQueried, timeCovered);
+            iter.remove();
+            cubeql.addCandidatePruningMsg(candidate,
+              new CandidateTablePruneCause(CandidateTablePruneCause.CandidateTablePruneCode.LESS_DATA));
+          }
+        }
+      }
+    }
+    //  cubeql.pruneCandidateFactWithCandidateSet(CandidateTablePruneCause.lessData(null));
 
   }
-  public MaxCoveringFactResolver(Configuration conf) {
+
+  private void resolveByDataCompleteness(CubeQueryContext cubeql) {
+    // From the list of  candidate fact sets, we calculate the maxDataCompletenessFactor.
+    float maxDataCompletenessFactor = 0f;
+    for (Candidate cand : cubeql.getCandidates()) {
+      float dataCompletenessFactor = computeDataCompletenessFactor(cand);
+      if (dataCompletenessFactor > maxDataCompletenessFactor) {
+        maxDataCompletenessFactor = dataCompletenessFactor;
+      }
+    }
+
+    if (maxDataCompletenessFactor == 0f) {
+      //there is nothing to prune
+      return;
+    }
+
+    // We prune those candidate fact set, whose dataCompletenessFactor is less than maxDataCompletenessFactor
+    Iterator<Candidate> iter = cubeql.getCandidates().iterator();
+    while (iter.hasNext()) {
+      Candidate cand = iter.next();
+      float dataCompletenessFactor = computeDataCompletenessFactor(cand);
+      if (dataCompletenessFactor < maxDataCompletenessFactor) {
+        log.info("Not considering Candidate :{} from the list as the dataCompletenessFactor for this:{} is "
+          + "less than the max:{}", cand, dataCompletenessFactor, maxDataCompletenessFactor);
+        iter.remove();
+        cubeql.addCandidatePruningMsg(cand,
+          new CandidateTablePruneCause(CandidateTablePruneCause.CandidateTablePruneCode.INCOMPLETE_PARTITION));
```

Comment:
use factory method CandidateTablePruneCause#incompletePartitions instead of  new CandidateTablePruneCause(..)